include/aspect/boundary_composition/function.h

```diff
@@ -0,0 +1,134 @@
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements boundary composition based on a functional
+     * description provided in the input file.
+     *
+     * @ingroup BoundaryCompositions
+     */
+    template <int dim>
+    class Function : public Interface<dim>, public SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Constructor.
+         */
+
+        /**
+         * Return the boundary composition as a function of position and time.
+         *
+         * @copydoc aspect::BoundaryComposition::Interface::boundary_composition()
+         */
+        virtual
+        double boundary_composition (const types::boundary_id boundary_indicator,
+                                     const Point<dim> &position, const unsigned int n_comp) const;
+        /**
+         * A function that is called at the beginning of each time step to
+         * indicate what the model time is for which the boundary values will
+         * next be evaluated. For the current class, the function passes to
+         * the parsed function what the current time is.
+         */
+        virtual
+        void
+        update ();
+
+        /**
+         * Return the minimal composition on that part of the boundary on
+         * which Dirichlet conditions are posed. Note: This is left over from
+         * temperature formulations, and may need to be removed later if its
+         * efficacy is not proven.
+         */
+        virtual
+        double minimal_composition (const std::set<types::boundary_id> &fixed_boundary_ids) const;
+
+        /**
+         * Return the maximal composition on that part of the boundary on
+         * which Dirichlet conditions are posed. Note: This is left over from
+         * temperature formulations, and may need to be removed later if its
+         * efficacy is not proven.
+         */
+        virtual
+        double maximal_composition (const std::set<types::boundary_id> &fixed_boundary_ids) const;
+
+        /**
+         * Declare the parameters this class takes through input files. The
+         * default implementation of this function does not describe any
+         * parameters. Consequently, derived classes do not have to overload
+         * this function if they do not take any runtime parameters.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         * The default implementation of this function does not read any
+         * parameters. Consequently, derived classes do not have to overload
+         * this function if they do not take any runtime parameters.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+        /**
+         * A function object representing the composition.
+         */
+        Functions::ParsedFunction<dim> boundary_composition_function;
```

Comment:
this variable is not used in the .cc file, you only use function. Please remove the variable and the comment.